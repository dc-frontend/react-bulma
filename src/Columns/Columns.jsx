import React from 'react'
import PropTypes from 'prop-types'

/**
 * Column
 * @param {*} props
 */
export function Columns (props) {
  return (
    <div className='columns'>
      {props.children}
    </div>
  )
}
Columns.propTypes = {
  children: PropTypes.any
}

/**
 * Column
 * @param {*} props
 */
export function Column (props) {
  const classes = props.className ? `column ${props.className}` : 'column'
  return (
    <div className={classes}>
      {props.children}
    </div>
  )
}
Column.propTypes = {
  children: PropTypes.any,
  class: PropTypes.any
}
