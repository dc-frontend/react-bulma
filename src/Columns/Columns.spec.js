// Link.react.test.js
import React from 'react'
import { Column, Columns } from './Columns'
import renderer from 'react-test-renderer'
import { cleanup } from '@testing-library/react'

afterEach(cleanup)

test('It renders as expected', () => {
  const component = renderer.create(
    <Columns>
      <Column>I am col 1</Column>
      <Column>I am col 2</Column>
      <Column>I am col 3</Column>
    </Columns>
  )
  expect(component.toJSON()).toMatchSnapshot()
})

test('Column can have custom class passed through', () => {
  const component = renderer.create(
    <Columns>
      <Column class='is-one-third'>I am col 1</Column>
    </Columns>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
