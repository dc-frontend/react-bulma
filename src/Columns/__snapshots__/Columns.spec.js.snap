// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders as expected 1`] = `
<div
  className="columns"
>
  <div
    className="column"
  >
    I am col 1
  </div>
  <div
    className="column"
  >
    I am col 2
  </div>
  <div
    className="column"
  >
    I am col 3
  </div>
</div>
`;
